paimon-core/src/main/java/org/apache/paimon/utils/BranchManager.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.utils;
+
+import org.apache.paimon.Snapshot;
+import org.apache.paimon.fs.FileIO;
+import org.apache.paimon.fs.Path;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+import static org.apache.paimon.schema.SchemaManager.SCHEMA_PREFIX;
+import static org.apache.paimon.utils.Preconditions.checkArgument;
+import static org.apache.paimon.utils.SnapshotManager.SNAPSHOT_PREFIX;
+import static org.apache.paimon.utils.TagManager.TAG_PREFIX;
+
+/** Manager for {@code Branch}. */
+public class BranchManager {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BranchManager.class);
+
+    private static final String BRANCH_PREFIX = "branch-";
+
+    private final FileIO fileIO;
+    private final Path tablePath;
+
+    public BranchManager(FileIO fileIO, Path path) {
+        this.fileIO = fileIO;
+        this.tablePath = path;
+    }
+
+    /** Return the root Directory of branch. */
+    public Path branchDirectory() {
+        return new Path(tablePath + "/branch");
+    }
+
+    /** Return the path of a branch. */
+    public String getBranchPath(String branchName) {
+        return tablePath + "/branch/" + BRANCH_PREFIX + branchName;
+    }
+
+    public Path branchPath(String branchName) {
+        return new Path(getBranchPath(branchName));
+    }
+
+    public Path tagPath(String tagName) {
+        return new Path(tablePath + "/tag/" + TAG_PREFIX + tagName);
+    }
+
+    public Path branchTagPath(String branchName, String tagName) {
+        return new Path(getBranchPath(branchName) + "/tag/" + TAG_PREFIX + tagName);
+    }
+
+    public Path snapshotPath(long snapshotId) {
+        return new Path(tablePath + "/snapshot/" + SNAPSHOT_PREFIX + snapshotId);
+    }
+
+    public Path branchSnapshotPath(String branchName, long snapshotId) {
+        return new Path(getBranchPath(branchName) + "/snapshot/" + SNAPSHOT_PREFIX + snapshotId);
+    }
+
+    public Path schemaPath(long schemaId) {
+        return new Path(tablePath + "/schema/" + SCHEMA_PREFIX + schemaId);
+    }
+
+    public Path branchSchemaPath(String branchName, long schemaId) {
```

Comment:
We should not build tag, snapshot and schema path in `BranchManager`

Reply:
done. Thanks for your review @FangYongs